Ginger/Ginger/Drivers/DriversWindows/MobileDriverWindow.xaml.cs

```diff
@@ -190,6 +317,309 @@ private async void MobileDriverWindow_DriverMessageEvent(object sender, DriverMe
+                    }
+                }
+
+                if (mDeviceGeneralInfo.TryGetValue("realDisplaySize", out value))
+                {
+                    if (string.IsNullOrEmpty((string)value))
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Screen resolution:", "N/A", DeviceInfo.eCategory.DeviceDetails));
+                    }
+                    else
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Screen resolution:", (string)value, DeviceInfo.eCategory.DeviceDetails));
+                    }
+                }
+
+                if (mDeviceGeneralInfo.TryGetValue("displayDensity", out value))
+                {
+                    if (string.IsNullOrEmpty((string)value.ToString()))
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Display density:", "N/A", DeviceInfo.eCategory.DeviceDetails));
+                    }
+                    else
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Display density:", (string)value.ToString(), DeviceInfo.eCategory.DeviceDetails));
+                    }
+                }
+
+                if (mDeviceGeneralInfo.TryGetValue("carrierName", out value))
+                {
+                    if (string.IsNullOrEmpty((string)value))
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Carrier name:", "N/A", DeviceInfo.eCategory.DeviceDetails));
+                    }
+                    else
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Carrier name:", (string)value, DeviceInfo.eCategory.DeviceDetails));
+                    }
+                }
+
+                if (mDeviceGeneralInfo.TryGetValue("bluetooth", out value))
+                {
+                    if (((Dictionary<string, object>)value).Count > 0)
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Bluetooth:", ((Dictionary<string, object>)value)["state"].ToString(), DeviceInfo.eCategory.DeviceDetails));
+                    }
+                    else
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Bluetooth:", "N/A", DeviceInfo.eCategory.DeviceDetails));
+                    }
+                }
+
+                if (mDeviceBatteryInfo.TryGetValue("power", out battery))
+                {
+                    mDeviceDetails.Add(new DeviceInfo("Device Battery:", battery + "%", DeviceInfo.eCategory.DeviceMetrics));
+                }
+                else
+                {
+                    mDeviceDetails.Add(new DeviceInfo("Battery:", "N/A", DeviceInfo.eCategory.DeviceMetrics));
+                }
+
+            }
+            else
+            {
+                mDeviceDetails.Add(new DeviceInfo("Device ID:", "N/A", DeviceInfo.eCategory.DeviceDetails));
```

Comment:
this else is not needed, just causing duplicated list to maintain, you already checking for each field if it exist and if not setting it to "N/A" so why again?